src/main/java/seedu/address/ui/PersonCard.java

```diff
@@ -57,9 +57,34 @@ public PersonCard(Person person, int displayedIndex) {
         address.setText(person.getAddress().value);
         email.setText(person.getEmail().value);
         description.setText(person.getDescription().value);
-        nextOfKin.setText(person.getNextOfKin().value);
+
+        // Set text for nextOfKin label if value is present, otherwise make it invisible
+        person.getNextOfKin().ifPresentOrElse(
+                nok -> {
+                    nextOfKin.setText(nok.value);
```

Comment:
You can abstract this out, similarly to `hideNode`

Reply:
However for this you might need to use `Label` instead of `Node` due to the `setText` method